### PR TITLE
Silence target down

### PR DESCRIPTION
**Pull request by Avi-Robusta**
To test with coredns with our stack
`robusta playbooks trigger prometheus_alert alert_name="TargetDown" namespace="kube-system" job_name="coredns" service="robusta-kube-prometheus-st-coredns" description="100% of the coredns/robusta-kube-prometheus-st-coredns targets in kube-system namespace are down." severity="warning"
`
To test with kube-dns with our stack
`robusta playbooks trigger prometheus_alert alert_name="TargetDown" namespace="kube-system" job_name="kube-dns" service="robusta-kube-prometheus-st-coredns" description="100% of the coredns/robusta-kube-prometheus-st-coredns targets in kube-system namespace are down." severity="warning"
`
To test with missing service with our stack
`robusta playbooks trigger prometheus_alert alert_name="TargetDown" namespace="kube-system" job_name="kube-dns" service="does-not-exist" description="100% of the coredns/robusta-kube-prometheus-st-coredns targets in kube-system namespace are down." severity="warning"
`

### CODE REVIEW

**Comment by Avi-Robusta** [comment]
add silence action option to print log

File: src/robusta/integrations/prometheus/models.py [hunk 1]
```diff
@@ -77,6 +77,12 @@ def get_deployment(self) -> Optional[RobustaDeployment]:
     def get_job(self) -> Optional[RobustaJob]:
         return self.job
 
+    def get_alert_label(self, label: str) -> Optional[str]:
+        try:
```
Comment (arikalon1) [inline src/robusta/integrations/prometheus/models.py:81]: I would write:
```
self.alert.labels.get(label, None)
```

File: playbooks/robusta_playbooks/targetdown_enrichment.py [hunk 2]
```diff
@@ -1,16 +1,37 @@
...
-def target_down_dns_enricher(alert: PrometheusKubernetesAlert, params: TimedPrometheusParams):
+def target_down_dns_enricher(alert: PrometheusKubernetesAlert):
     """
     Enrich the finding with a detailed explanation for the cause of the CoreDNS and
     Kube-DNS unreachable
     """
-    if not alert.job or alert.job not in ["corens", "kube-dns"]:
+    job = alert.get_alert_label('job')
+    service = alert.get_alert_label('service')
+    if not job or job not in ["coredns", "kube-dns"]:
```
Comment (arikalon1) [inline playbooks/robusta_playbooks/targetdown_enrichment.py:34]: Job is either None or some str value.
Is it enough to write:
```
if job not in ["coredns", "kube-dns"]:
```

**Review by arikalon1 — APPROVED** [review]
LGTM
2 minor comments

File: playbooks/robusta_playbooks/alerts_integration.py [hunk 3]
```diff
@@ -66,6 +65,24 @@ class NameSilencerParams(ActionParams):
...
+class SilenceAlertParams(ActionParams):
+    """
+    :var log_silence: prints a log in robusta logs of the silence.
+    """
+
+    log_silence: bool = False
+
+
+@action
+def silence_alert(alert: PrometheusKubernetesAlert, params: SilenceAlertParams):
```
Comment (Avi-Robusta) [inline playbooks/robusta_playbooks/alerts_integration.py:77]: @arikalon1 LMK if you can think of anything else to add here

Comment (arikalon1) [reply playbooks/robusta_playbooks/alerts_integration.py:77]: it's perfect

**Review by arikalon1 — APPROVED** [review]
LGTM

**Pull request merged**